src/CommunityPatch/Patches/Feats/AseraiCheapCaravansPatch.cs

```diff
@@ -0,0 +1,87 @@
+        
+        private static readonly Type TargetType = AccessTools.TypeByName("SandBox.LordConversationsCampaignBehavior");
+
+        private static readonly MethodInfo TargetMethodInfo = AccessTools.Method(TargetType, "conversation_magistrate_form_a_caravan_accept_on_consequence");
+        private static readonly MethodInfo PrefixPatchMethodInfo = AccessTools.Method(typeof(AseraiCheapCaravansPatch), nameof(Prefix));
+        private static readonly MethodInfo PostfixPatchMethodInfo = AccessTools.Method(typeof(AseraiCheapCaravansPatch), nameof(Postfix));
+
+        private static readonly byte[][] TargetHashes = {
+            new byte[] {
+                // e1.2.0.226271
+                0x72, 0x71, 0xCF, 0x96, 0x32, 0xA5, 0x98, 0x86,
+                0xAB, 0x61, 0xDD, 0x80, 0x8E, 0x36, 0xCC, 0xF2,
+                0x77, 0x37, 0x51, 0xC3, 0x5B, 0x6A, 0x46, 0xF6,
+                0x22, 0x1B, 0x88, 0xED, 0x36, 0xFB, 0xAA, 0x7A
+            }
+        };
+
+        public override IEnumerable<MethodBase> GetMethodsChecked() {
+            yield return AseraiCheapCaravansPatch.TargetMethodInfo;
+        }
+        
+        public override bool? IsApplicable(Game game) {
+            // This patch does not modify the target function with Prefix, which only provides state data for Postfix
+            // It should be compatible with other patches
+
+            var hash = TargetMethodInfo.MakeCilSignatureSha256();
+            return hash.MatchesAnySha256(TargetHashes);
+        }
+        
+        public override void Apply(Game game) {
+            if (Applied) return;
+
+            CommunityPatchSubModule.Harmony.Patch(TargetMethodInfo,
+                prefix: new HarmonyMethod(PrefixPatchMethodInfo),
+                postfix: new HarmonyMethod(PostfixPatchMethodInfo));
+                
+                Applied = true;
+        }
+        
+        public override void Reset() { }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void Prefix(out int __state) {
+            // get player gold and make it available for Postfix
+            __state = Hero.MainHero.Gold;
+        }
+
+        static void Postfix(int __state) {
+            // presume the player successfully purchased a caravan if their gold is less than it was at the beginning
+            // of the target function call
+            if (Hero.MainHero.Gold < __state &&
+                Hero.MainHero.IsHumanPlayerCharacter &&
+                Hero.MainHero.CharacterObject.GetFeatValue(DefaultFeats.Cultural.AseraiCheapCaravans)) {
+                
+                int creditAmount = 0;
+                if (DefaultFeats.Cultural.AseraiCheapCaravans.EffectBonus == 0f) {
+                    // As of e1.2.0.226271, Aserai bonus is not set
+                    // Assume 30% caravan discount per character creation screen text
+                    creditAmount = (int) (0.30f * (__state - Hero.MainHero.Gold));
```

Comment:
Could be cool if you tried to parse it from the description, then fallback on 30%
```suggestion
                    // try to get it from the feat description, or assume 30% discount as per what the
                    // character creation screen text at one time said.
                    float discount = 0.30f;
      	            try {
                        Match featDescMatch = Regex.Match(DefaultFeats.Cultural.AseraiCheapCaravans.Description.ToString(),
                                                          @"Caravans are ([\d\.]+)% cheaper to build", RegexOptions.IgnoreCase);
      	                if (featDescMatch.Success) { 
							              discount = float.Parse(featDescMatch.Groups[1].Value, CultureInfo.InvariantCulture) / 100.0f;
                        }
                    } catch (FormatException) {}
                    creditAmount = (int) (discount * (__state - Hero.MainHero.Gold));
```

Reply:
would also require at the top of file:

```
using System.Globalization;
using System.Text.RegularExpressions;
```

Reply:
Is Description a TextObject? ToString will localize it if it's not English.

Reply:
> This .3 doesn't exist somewhere else, like the Perk's primary bonus?
> `if (1 - DefaultFeats.Cultural.AseraiCheapCaravans.EffectBonus)` gives the .3, that should be expressed as a property in the patch and reused, most likely.

This is inside an `if` branch where the [EffectBonus is 0](https://github.com/Tyler-IN/MnB2-Bannerlord-CommunityPatch/pull/215/files/93fb699b2b7fb454abba5a9362306037594b0e95#diff-455508814d7cc07863450716b5a3e047R68) which he assumed to be wrong and substituted a different value. Indeed if the [EffectBonus exists](https://github.com/Tyler-IN/MnB2-Bannerlord-CommunityPatch/pull/215/files/93fb699b2b7fb454abba5a9362306037594b0e95#diff-455508814d7cc07863450716b5a3e047R75) it is used.

> Is Description a TextObject? ToString will localize it if it's not English.
> 

Yes Description is a TextObject. I'll see if I can get the raw text to avoid the localization, but it's only a fallback. Code uses EffectBonus if available.

Reply:
Sadly, looking over TextObject.cs, I don't see a way to avoid the localization.

Reply:
The 0.3f bonus isn't defined anywhere else at the moment for this feat

Reply:
Although you could get the unlocalized string from the TextObject with reflection. But i'll leave this optional for now. It's just a nifty little thing.

Reply:
There's a trick I've done elsewhere, but I don't think we should use it.
```cs
      var textObjStrings = TextObject.ConvertToStringList(
        new List<TextObject> {
          _perk.Name,
          _perk.Description
        }
      );
```
This extracts `Value` field values directly from TextObjects.

Reply:
Cool, i'll remember that. 👍

(when there's no other choice)